src/composer/src/build.rs

```diff
@@ -371,14 +377,46 @@ impl BuildState for DefaultBuilder {
     fn comp_obj_path(&self, c: &ComponentId, s: &SystemState) -> Result<String, String> {
         self.comp_file_path(&c, &self.comp_obj_file(&c, &s), &s)
     }
+    
+    fn comp_const_h(
+        &self, id: &ComponentId, s: &SystemState, header_file: &mut Option<String>
+    ) -> Result<(), String> {
+        let c = component(&s, id);
+        let header_file_path = self.comp_file_path(id, &"component_constants.h".to_string(), &s)?;
+
+        if std::path::Path::new(&header_file_path).exists() {
+            //check the existing header file for constructor_build()
+            *header_file = Some(header_file_path);
+            return Ok(());
+        }
+
+        if !c.constants.is_empty() {    
+            let mut header_content = String::from("#ifndef COMPONENT_CONSTANTS_H\n#define COMPONENT_CONSTANTS_H\n\n");
+            
+            for constant in &c.constants {
+                header_content.push_str(&format!("#define {} {}\n", constant.variable, constant.value));
+            }
+    
+            header_content.push_str("\n#endif // COMPONENT_CONSTANTS_H\n");
+        
+            emit_file(&header_file_path, header_content.as_bytes()).unwrap();
+
+            *header_file = Some(header_file_path);
+        }
+
+        Ok(())
+    }
 
     fn comp_build(&self, id: &ComponentId, state: &SystemState) -> Result<String, String> {
         let comp_dir = self.comp_dir_path(&id, &state)?;
         compdir_check_build(&comp_dir)?;
         let p = state.get_param_id(&id);
         let output_path = self.comp_obj_path(&id, &state)?;
 
-        let cmd = comp_gen_make_cmd(&output_path, p.param_prog(), p.param_fs(), &id, &state);
+        let mut header_file: Option<String> = None;
```

Comment:
Given the change above to the return value, you'll be able to remove the `mut` and make this a String, not an option. It can directly get the return value from the function call.